developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -1671,6 +1674,10 @@ end
 To access the value of a setting, use `Setting[:example_setting]` from
 anywhere in your plugin.
 
+The settings are strongly typed and you have to define it.
+The basic types supported by Foreman are: `:boolean`, `:integer`, `:float`, `:string`, `:text`, `:hash`, `:array`.
```

Comment:
Should we add a few other common types, such as `email`, `fqdn` or `url`? that would also allow having UI validations for them

Reply:
I'm not sure it adds something more then `type: :string, validate: :email`.
Though I'm not sure this is the place for it. I'd add types once we want to introduce use-cases for them.
It would change only very little amount of settings, so I don't think it's necessary to have from day 0.

Reply:
my thought was it could make it easier to add a client-side component that handles validation for it for some of the more common types, and get rid of some of the validations that are currently handled by constants, but not crucial.